runtime/commonMain/src/kotlinx/serialization/internal/Tuples.kt

```diff
@@ -6,179 +6,160 @@ package kotlinx.serialization.internal
+    override val descriptor: SerialDescriptor get() = TripleDesc
 
     override fun serialize(encoder: Encoder, obj: Triple<A, B, C>) {
-        @Suppress("NAME_SHADOWING")
-        val encoder = encoder.beginStructure(descriptor, aSerializer, bSerializer, cSerializer)
-        encoder.encodeSerializableElement(descriptor, 0, aSerializer, obj.first)
-        encoder.encodeSerializableElement(descriptor, 1, bSerializer, obj.second)
-        encoder.encodeSerializableElement(descriptor, 2, cSerializer, obj.third)
-        encoder.endStructure(descriptor)
+        val structuredEncoder = encoder.beginStructure(descriptor, aSerializer, bSerializer, cSerializer)
+        structuredEncoder.encodeSerializableElement(descriptor, 0, aSerializer, obj.first)
+        structuredEncoder.encodeSerializableElement(descriptor, 1, bSerializer, obj.second)
+        structuredEncoder.encodeSerializableElement(descriptor, 2, cSerializer, obj.third)
+        structuredEncoder.endStructure(descriptor)
     }
 
     override fun deserialize(decoder: Decoder): Triple<A, B, C> {
-        @Suppress("NAME_SHADOWING")
-        val input = decoder.beginStructure(descriptor, aSerializer, bSerializer, cSerializer)
-        var aSet = false
-        var bSet = false
-        var cSet = false
-        var a: Any? = null
-        var b: Any? = null
-        var c: Any? = null
+        val composite = decoder.beginStructure(descriptor, aSerializer, bSerializer, cSerializer)
+        if (composite.decodeSequentially()) {
+            return decodeSequentially(composite)
+        }
+        return decodeStructure(composite)
+    }
+
+    private fun decodeSequentially(composite: CompositeDecoder): Triple<A, B, C> {
+        val a = composite.decodeSerializableElement(descriptor, 0, aSerializer)
+        val b = composite.decodeSerializableElement(descriptor, 1, bSerializer)
+        val c = composite.decodeSerializableElement(descriptor, 2, cSerializer)
+        composite.endStructure(descriptor)
+        return Triple(a, b, c)
+    }
+
+    private fun decodeStructure(composite: CompositeDecoder): Triple<A, B, C> {
+        var a: Any? = NULL
+        var b: Any? = NULL
+        var c: Any? = NULL
         mainLoop@ while (true) {
-            when (input.decodeElementIndex(descriptor)) {
-                CompositeDecoder.READ_ALL -> {
-                    a = input.decodeSerializableElement(descriptor, 0, aSerializer)
-                    aSet = true
-                    b = input.decodeSerializableElement(descriptor, 1, bSerializer)
-                    bSet = true
-                    c = input.decodeSerializableElement(descriptor, 2, cSerializer)
-                    cSet = true
-                    break@mainLoop
-                }
+            when (val index = composite.decodeElementIndex(descriptor)) {
                 CompositeDecoder.READ_DONE -> {
                     break@mainLoop
                 }
                 0 -> {
```

Comment:
Redundant curly braces

Reply:
I personally find consistent braces in a single `when` much more readable